### PR TITLE
Make `assigns.myself` a struct to catch mistakes

### PR DESCRIPTION
**Pull request by henrik**
See #895.

Thought it could be helpful to see what the proposed fix looks like, and if it's still something you want to go ahead with. (And an opportunity for me to learn LiveView a bit better.)

### CODE REVIEW

File: test/phoenix_live_view/diff_test.exs [hunk 1]
```diff
@@ -590,7 +591,7 @@ defmodule Phoenix.LiveView.DiffTest do
       assert {MyComponent, "hello", _, _, _} = cid_to_component[0]
 
       assert_received {:mount, %Socket{endpoint: __MODULE__, assigns: assigns}}
-                      when assigns == %{flash: %{}, myself: 0}
+                      when assigns == %{flash: %{}, myself: %CID{cid: 0}}
```
Comment (henrik) [inline test/phoenix_live_view/diff_test.exs:594]: Not sure how you feel about this. Figured I'd use the full module name in most places for clarity, but it got a bit noisy in this test.

**Comment by chrismccord** [comment]
Looks perfect from my side. I'll let @josevalim take a pass as well. Thanks!

**Comment by josevalim** [comment]
Fantastic, thanks for the contributions, very appreciated! :green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by henrik** [comment]
It’s a fun project to be a part of! Thanks for all the work the both of you put in ❤️

**Pull request merged**